llarp/config/config.cpp

```diff
@@ -1132,18 +1134,24 @@ namespace llarp
     if (ec)
       throw std::runtime_error(stringify("filesystem error: ", ec));
 
+    const auto parent = confFile.parent_path();
+
     // create parent dir if it doesn't exist
-    if (not fs::exists(confFile.parent_path(), ec))
+    if ((not parent.empty()) and (not fs::exists(parent, ec)))
     {
-      if (not fs::create_directory(confFile.parent_path()))
+      if (not fs::create_directory(parent))
         throw std::runtime_error(stringify("Failed to create parent directory for ", confFile));
```

Comment:
this should be something more like "Unable to create config file: unable to create directory \"directory_path\" and it does not exist."